### PR TITLE
PIM-11013 : Fix edit user profile without password changes

### PR DESCRIPTION
**Pull request by jeremyBeaucousinAkeneo**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/UserManagement/back/Application/Command/UpdateUserCommand/UpdateUserCommandHandler.php [hunk 1]
```diff
@@ -116,6 +116,8 @@ public function handle(UpdateUserCommand $updateUserCommand): UserInterface
 
     private function isPasswordUpdating($data): bool
     {
-        return array_key_exists('current_password', $data) || array_key_exists('new_password', $data) || array_key_exists('new_password_repeat', $data);
+        return (isset($data['current_password']) && !empty($data['current_password'])) ||
+            (isset($data['new_password']) && !empty($data['new_password'])) ||
+            (isset($data['new_password_repeat']) && !empty($data['new_password_repeat']));
```
Comment (pchasle) [inline src/Akeneo/UserManagement/back/Application/Command/UpdateUserCommand/UpdateUserCommandHandler.php:121]: `isset` is redundant with `empty` (see https://www.php.net/manual/en/function.empty.php)

This should be enough:
```
        return !empty($data['current_password'])
            || !empty($data['new_password'])
            || !empty($data['new_password_repeat']);
```

And are you sure that it shouldn't be a `&&` condition to check that all 3 values are present to do a password update?

Comment (pchasle) [reply src/Akeneo/UserManagement/back/Application/Command/UpdateUserCommand/UpdateUserCommandHandler.php:121]: Or maybe the fact that it's `||` allow to raise validation error to the frontend when the user is editing the form :thinking:

**Review by pchasle — APPROVED** [review]

File: src/Akeneo/UserManagement/back/Application/Command/UpdateUserCommand/UpdateUserCommandHandler.php (hunk 1 above)
Comment (jeremyBeaucousinAkeneo) [reply src/Akeneo/UserManagement/back/Application/Command/UpdateUserCommand/UpdateUserCommandHandler.php:121]: I put back the old rile as it was implemented before the [PIM-10639](https://github.com/akeneo/pim-community-dev/pull/19651/files#top)  : (cf : https://github.com/akeneo/pim-community-dev/pull/19651/files)

```php
private function isPasswordUpdating($data): bool
    {
        return (isset($data['current_password']) && !empty($data['current_password'])) ||
            (isset($data['new_password']) && !empty($data['new_password'])) ||
            (isset($data['new_password_repeat']) && !empty($data['new_password_repeat']));
    }
```

@amigaire has warned about the potential bound effect of modifying, the current ticket is one that was not identified.

Comment (pchasle) [reply src/Akeneo/UserManagement/back/Application/Command/UpdateUserCommand/UpdateUserCommandHandler.php:121]: I agree, was just wondering.

**Review by amigaire — APPROVED** [review]

**Pull request merged**